src/lolex-src.js

```diff
@@ -561,6 +574,32 @@ function withGlobal(_global) {
             return [secsSinceStart, remainderInNanos];
         }
 
+        clock.requestIdleCallback = function requestIdleCallback(func, timeout) {
+            var timeToNextIdlePeriod = 0;
+
+            if (clock.countTimers() > 0) {
+                timeToNextIdlePeriod = 50; // const for now
+            }
+
+            if (typeof timeout === "undefined") {
+                timeout = timeToNextIdlePeriod;
+            } else {
+                timeout = Math.min(timeout, timeToNextIdlePeriod);
+            }
+
+            var result = addTimer(clock, {
+                func: func,
+                args: Array.prototype.slice.call(arguments, 2),
+                delay: timeout
```

Comment:
I am not a fan of assigning to arguments, as that is only safe for primitives (pass by value).
I am also not a fan of using an `if-else` for a conditional assignment, as people sometimes add more statements (side effects).

Instead, I'd suggest avoiding assigning to `timeout` altogether, and calculating the correct value for `delay` using the ternary operator.

```suggestion
                delay: typeof timeout === "undefined" ? timeToNextIdlePeriod : Math.min(timeout, timeToNextIdlePeriod)
```